sqlglot/dataframe/sql/dataframe.py

```diff
@@ -365,20 +365,27 @@ def select(self, *cols, **kwargs) -> DataFrame:
                     x.this for x in get_tables_from_expression_with_join(self.expression)
                 ]
                 cte_names_in_join = [x.this for x in join_table_identifiers]
+                # If we a columns resolves to multiple CTE expressions then we want to use each CTE left-to-right
+                # and therefore we allow multiple columns with the same name in the result. This matches the behavior
+                # of Spark.
+                resolved_column_position: t.Dict[str, int] = {}
                 for ambiguous_col in ambiguous_cols:
                     ctes_with_column = [
                         cte
                         for cte in self.expression.ctes
                         if cte.alias_or_name in cte_names_in_join
```

Comment:
you should do this outside of the loop, and probably make a set

for cte in self.expression.ctes if cte.alias_or_name in cte_names in join

Reply:
in general this code isn't really correct, it should really be using scopes but i guess we can leave that for another day

Reply:
What do you mean by incorrect? This code was written prior to scopes so that is why it isn't currently used but I believe this functions correctly. There are likely a few places that would benefit from scopes now that they exist.